spec/src/main/asciidoc/Structure and format.adoc

```diff
@@ -0,0 +1,81 @@
+
+==== Property key encoding
+
+Since an individual _configuration key_ may contain a dot (`.`) character, using a `.` character to separate `property` keys into a _configuration path_ poses a problem.
+Because of this, an encoding scheme should be used when dealing with keys that contain `.` characters.
+
+When properties are loaded from a `properties` file or from the system `properties` map, each encountered key is transformed into a _configuration path_ as follows:
+
+* Any backslash (`\`) character found in the `property` key is considered to be an _escape_ for the next character
+* Any character which is immediately preceded by an _escape_ is added to the configuration key verbatim
+* Any dot (`.`) character which is _not_ immediately preceded by an _escape_ marks a separate _configuration key_ in the _configuration path_
+
+The implementation *must* reject `properties` files or sets which contain a key that ends with an _escape_ backslash (`\`).
+
+*Note:* The `properties` format also escapes backslashes, so when looking inside a `properties` file, a dot (`.`) which is embedded within a configuration key will appear to be encoded as `\\.`, and a backslash (`\`) which is embedded within a configuration key will appear to be encoded as `\\\\`.
+
+The encoded property name is then looked up in the corresponding `properties` file, and the corresponding value (if any) is used as the _raw value_ for that configuration key.
+
+==== TODO: List key encoding per previous discussions
+
+==== TODO: Value encoding (esp. lists, complex types, cross-references, etc.)
+
+The `properties` value string is used as the _raw_ value of the corresponding configuration.
+[...etc...]
+
+=== Mapping to YAML format
+
+The YAML file format is defined by https://yaml.org/spec/1.2.2/[a specification].
+A YAML file is recognized as a configuration source by this specification, according to the following rules:
+
+* A _configuration entry_ corresponds to a YAML https://yaml.org/spec/1.2.2/#mapping[mapping], with the _mapping key_ being used as the _configuration key_ and the _mapping value_ as the nested _configuration_.
+* TODO: YAML sequences are lists...
+* The _raw value_ for a given _configuration_ is produced by creating a _canonical representation_ of the scalar value of the property, mapped according to the YAML _tag_ of the value:
+    * TODO: lists again...?
+    * A value which has a YAML _tag_ of `tag:yaml.org,2002:null` corresponds to a `null` (missing) _raw value_.
+    * A value which has a YAML _tag_ of `tag:yaml.org,2002:str` is directly used as a _raw value_.
+    * A value which has a YAML _tag_ of `tag:yaml.org,2002:bool` is parsed according to the YAML parsing rules for that type; `true` values are then represented as the string `"true"`, and `false` values are represented as the string `"false"`.
+    * TODO: FP types canonical form
+    * TODO: Integer canonical form
+    * TODO: Custom tags for conversion types...?
+    * ...etc...
```

Comment:
Hopefully this adequately captures the conclusion of the discussion we had about "native" type conversion. The idea is to parse the values according to the rules of YAML, and then canonicalize them into strings in a type-specific manner according to the canonical representation rules for each corresponding configuration converter.